### PR TITLE
further improvements to duration / checking live

### PR DESCRIPTION
**Pull request by squarebracket**
After speaking with some people in the dash-if slack, it sounds like the best method for determining if a feed is live is by checking both the `isDynamic` and checking the manifest to see if it has a duration.

### CODE REVIEW

File: src/js/videojs-dash.js [hunk 1]
```diff
@@ -171,6 +172,16 @@ class Html5DashJS {
 
     this.mediaPlayer_.on(dashjs.MediaPlayer.events.ERROR, this.retriggerError_);
 
+    this.getDuration_ = (event) => {
+      let periods = event.data.Period_asArray;
+
+      if (event.data.mediaPresentationDuration || periods[periods.length - 1].duration) {
```
Comment (gkatsev) [inline src/js/videojs-dash.js]: so, if a duration exists, it means this isn't live?

File: src/js/videojs-dash.js [hunk 2]
```diff
@@ -261,7 +273,7 @@ class Html5DashJS {
   }
 
   duration() {
-    if (this.mediaPlayer_.isDynamic()) {
+    if (this.mediaPlayer_.isDynamic() && !this.hasFiniteDuration_) {
```
Comment (gkatsev) [inline src/js/videojs-dash.js]: Is it possible for us to not update `this.hasFiniteDuration_` before `duration` is requested and thus we'll be stuck in live mode?

File: src/js/videojs-dash.js (hunk 1 above)
Comment (squarebracket) [reply src/js/videojs-dash.js]: Correct. I'll have to double check the spec, but I believe a duration on the currently-active period or overall presentation are mutually exclusive with being live content.

File: src/js/videojs-dash.js (hunk 2 above)
Comment (squarebracket) [reply src/js/videojs-dash.js]: I don't remember exactly where it is in video.js, but the thing that makes it live is a duration check in the control bar, right? If it's listening to `durationchange`, then we could fire one off when the manifest loads (i.e. in `getDuration`) to make sure the control bar always updates to be live/not live. That should cover any situation where we're streaming a live feed that becomes non-live.

Comment (squarebracket) [reply src/js/videojs-dash.js]: Ah yes, seems to be [here](https://github.com/videojs/video.js/blob/master/src/js/control-bar/live-display.js#L29). Should I add in that `trigger`?

File: src/js/videojs-dash.js (hunk 1 above)
Comment (squarebracket) [reply src/js/videojs-dash.js]: Ugh, the spec and IOP are too long and crazy for me to validate this easily, but this is what was suggested by one of the regulars in the dash-if slack.

File: src/js/videojs-dash.js (hunk 2 above)
Comment (gkatsev) [reply src/js/videojs-dash.js]: seems reasonable to add a durationchange trigger.

**Pull request merged**